Source/Reporters/JunitXMLReporter.jsl

```diff
@@ -33,7 +33,148 @@ Example of JunitXML format:
 
 Define Class(
   "UtJunitXMLReporter",
-  Base Class( UtCollectingReporter ),
+  Base Class( "UtReporter" ),
+
+  // Variable: hash_results
+  // associative array of the test results
+  // test suite => test case => successes, failures, throws
+  hash_results = [=>Associative Array()];
+
+  // Variable: n successes
+  n successes = 0;
+
+  // Variable: n failures
+  n failures = 0;
+
+  // Variable: n unexpected throws
+  n unexpected throws = 0;
+
+  base dir = "";
+
+  _init_ = Method({base dir},
+    If( NItems( Words(base dir, "/\") ) > 1,
+      Throw( "UtJunitXMLReporter requires base dir to be a single directory's name." );
+    );
+    this:base dir = If( Length(base dir) != 1,
+                        // Remove any separators
+                        Substitute( base dir, "\", "", "/", "" )
+                      , // special case for "/" or "\" to get a full path
+                        base dir
+                    );
+
+  );
+
+  /*
+  Method: add to hash
+
+  ---Prototype---
+  add to hash(curr_suite, curr_test, result)
+  ---------------
+  Adds the result of a test to hash_results
+
+  Arguments:
+    curr_suite - the name of the test suite
+    curr_test - the name of the test
+    result - the result of the test
+  */
+  add to hash = Method({curr_suite, curr_test, result},
+    If(Contains(this:hash_results << get keys, curr_suite),
+      If(Contains(this:hash_results[curr_suite] << get keys, curr_test),
+        Insert Into(this:hash_results[curr_suite][curr_test], result);
+        ,
+        this:hash_results[curr_suite][curr_test] = Eval List({result});
+      )
+      ,
+      result_list = Eval List({result});
+      this:hash_results[curr_suite] = Associative Array(Eval List({curr_test}), Eval List({result_list}));
+    );
+  );
+
+  /*
+  Method: get file path
+
+  ---Prototype---
+  get file path()
+  ---------------
+  Gets the file path that the test came from.
+  The path will be relative to the base dir if provided.
+  */
+  get file path = Method( {},
+    If( Include File List()[1] == "",
+      Return("")
+    );
+    full path = Convert File Path( Include File List()[1] );
+    If( this:base dir == "/" | this:base dir == "\",
+      Return(full path)
+    );
+    dirs = Words( full path, "/\");
+    path start = Contains( dirs, this:base dir );
+    If( path start,
+      Return( Concat Items( dirs[path start::NItems(dirs)], "/" ) );
+      Return( path );
+    ,
+      Return( Word( -1, full path, "/\" ) );
```

Comment:
`Return( path )` is unreachable. Also, I would remove the returns since this is the last line and will be return anyway ( this slightly helps performance).
```
Concat Items(...)
,
Word(...)
```

Reply:
Oops yup. I missed that with my last change. Thanks!